libs/core/src/components/sage-checkbox/test/sage-checkbox.spec.tsx

```diff
@@ -0,0 +1,134 @@
+    expect(input?.disabled).toBe(true);
+  });
+  it('renders with error when invalid prop is set', async () => {
+    const page = await newSpecPage({
+      components: [SageCheckbox],
+      html: `<sage-checkbox checkbox-id="default" label="Label text" invalid />`,
+    });
+    expect(page.root).toEqualHtml(`
+      <sage-checkbox checkbox-id="default" label="Label text" invalid>
+        <mock:shadow-root>
+          <div class="sage-checkbox is-invalid">
+            <input type="checkbox" id="default">
+            <label htmlfor="default">Label text</label>
+          </div>
+        </mock:shadow-root>
+      </sage-checkbox>
+    `);
+  });
+  it('renders indeterminate when indeterminate prop is set', async () => {
+    const page = await newSpecPage({
+      components: [SageCheckbox],
+      html: `<sage-checkbox checkbox-id="default" label="Label text" indeterminate />`,
+    });
+    expect(page.root).toEqualHtml(`
+      <sage-checkbox checkbox-id="default" label="Label text" indeterminate>
+        <mock:shadow-root>
+          <div class="sage-checkbox is-indeterminate">
+            <input type="checkbox" id="default">
+            <label htmlfor="default">Label text</label>
+          </div>
+        </mock:shadow-root>
+      </sage-checkbox>
+    `);
+  });
+  it('renders label text when label prop is set', async () => {
+    const page = await newSpecPage({
+      components: [SageCheckbox],
+      html: `<sage-checkbox checkbox-id="default" label="This is label text" />`,
+    });
+    expect(page.root).toEqualHtml(`
+      <sage-checkbox checkbox-id="default" label="This is label text">
+        <mock:shadow-root>
+          <div class="sage-checkbox">
+            <input type="checkbox" id="default">
+            <label htmlfor="default">This is label text</label>
+          </div>
+        </mock:shadow-root>
+      </sage-checkbox>
+    `);
+  });
+  it('renders message when message prop is set', async () => {
+    const page = await newSpecPage({
+      components: [SageCheckbox],
+      html: `<sage-checkbox checkbox-id="default" label="Label text" message="This is short message text." />`,
+    });
+    expect(page.root).toEqualHtml(`
+      <sage-checkbox checkbox-id="default" label="Label text" message="This is short message text.">
+        <mock:shadow-root>
+          <div class="sage-checkbox">
+            <input type="checkbox" id="default">
+            <label htmlfor="default">Label text</label>
+            <div class="sage-checkbox__message">This is short message text.</div>
+          </div>
+        </mock:shadow-root>
+      </sage-checkbox>
+    `);
+  });
+  it('renders required input when required prop is set', async () => {
+    const { root } = await newSpecPage({
+      components: [SageCheckbox],
+      html: `<sage-checkbox checkbox-id="default" label="Label text" required />`,
+    });
+    const input = root?.shadowRoot?.querySelector('input');
+    expect(input?.required).toBe(true);
```

Comment:
I have no issue with this approach but I'm curious why you chose to check it with `input?.required` instead of checking against the `page.root` HTML.

Reply:
no rhyme or reason really. I believe I saw it used in another component, but felt to me it was better to be able to check the input vs. just rendering out the markup.